tests/common/reboot.py

```diff
@@ -143,23 +150,27 @@ def execute_reboot_helper():
     logger.info('ssh has started up')
 
     logger.info('waiting for switch to initialize')
-    time.sleep(wait)
+
+
+    logger.info('waiting for warmboot-finalizer service to become activating')
+    finalizer_state = get_warmboot_finalizer_state(duthost)
+    while finalizer_state != 'activating':
+        time.sleep(1)
+        dut_datetime_after_ssh = duthost.get_up_time()
+        time_passed = float(dut_datetime_after_ssh.strftime("%s")) - float(dut_datetime.strftime("%s"))
+        if time_passed > wait:
+            raise Exception('warmboot-finalizer never reached state "activating"')
+        finalizer_state = get_warmboot_finalizer_state(duthost)
```

Comment:
I think:
1. this sleep could fail false positively if the reboot type is not warm.
2. there is no increment of wait time. If the original problem was that we didn't reach warm reboot finalizer 'starting' state, then this change didn't give more wait. Unless the original issue was that warm reboot finalizer finished before wait is over, then this change makes sense. But then it should be guarded with reboot type check.
3. I would put sleep 1 second above the last line. So that the last result will be checked.

Reply:
The original problem was not that we did not reach `activating` state. It was however that `activating` state was reached and already changed before we asserted it. This led to failed assert. Increasing wait time will only make this problem worse.

The issue https://github.com/Azure/sonic-mgmt/issues/2147 captures that the `finalizer_state` is already `inactive` when the code checks `assert finalizer_state == 'activating'`. This was due to dead sleep in the code where the `finalizer_state` changes from `activating` to `inactive` during the sleep.

Now, with this change, the dead-sleep is replaced by per second check of the state to be `activating`. If the state does not reach `activating` within `wait` time, the test fails by raising exception.

Reply:
I agree that the check should be done only for `reboot_type` as `warm`

Reply:
can we just simplify this? why not change the original assertion to check for activating and inactive? if the warm reboot didn't trigger, we would already catch it by checking if ssh conn didn't drop

Reply:
My thought behind this is to verify that `warmboot_finalizer` changes the state from `activating` (within `wait` time) to `inactive` (within `timeout` time).

I can simplify this by just replacing the existing check from:
`assert finalizer_state == 'activating'` to:
`assert (finalizer_state == 'activating' or finalizer_state == 'inactive'`). But this check will not be as strict in ensuring the state changes as expected.

Reply:
Agree. It is good to have the state transition check as well